src/vdxf/index.ts

```diff
@@ -116,9 +118,9 @@ export class VDXFObject implements VDXFObjectInterface {
     
     writer.writeVarInt(new BN(this.version, 10));
 
-    if (dataLength) {      
-      writer.writeVarSlice(this.toDataBuffer());
-    }
+    
+    writer.writeVarSlice(this.toDataBuffer());
```

Comment:
Also confused as to why zero data length objects are calling toDataBuffer and now moving offset by one, I might be missing something, will discuss.